src/yocto/recipes.md

```diff
@@ -0,0 +1,186 @@
+    SRC_URI = "file://helloworld.c" 
+    S = "${WORKDIR}" 
+    do_compile() { 
+    ${CC} ${CFLAGS} ${LDFLAGS} helloworld.c -o helloworld 
+    } 
+    do_install() { 
+    install -d ${D}${bindir} 
+    install -m 0755 helloworld ${D}${bindir} 
+    } 
+```
+# .bbappend file
+Recipes used to append Metadata to other recipes, there are calls append files, these files use the .bbappend file type. This file is used to modify the recipe file, this should have the same name that the recipe, at the same time bbappend files allow your layer to make additions or changes to the content of another layer's recipe without having to copy the other recipe into your layer.  
+Being able to append information to an existing recipe not only avoid duplication, also automatically applies recipe changes in a different layer to your layer.
+## Extending recipes with .bbappend files
+It is not necessary to recreate entire recipe files from scratch,  you can use **. bbappend** files to supplement an existing recipe file with new information, providing that the original information in the recipe file, resides in an existing layer.
+## Systemd recipes
+To use systemd services has necessary enabled this in the final image, because this is not enabled by default. Add the next lines to the configuration file **local. conf** to enable system as a default unit manager.
+
+```
+DISTRO_FEATURES_append = " systemd"
+DISTRO_FEATURES_BACKFILL_CONSIDERED += "sysvinit"
+VIRTUAL-RUNTIME_init_manager = "systemd"
+VIRTUAL-RUNTIME_initscripts = "systemd-compat-units"
+```
+
+### Important varibales for systemd recipe
+1.**inherit systemd**
+Each target has a name instead of a number, ensure the recipe inherits from the systemd class:  
+``` inherit systemd ```  
+2. NATIVE_SYSTEMD_SUPPORT = "1"
+3. **SYSTEMD_AUTO_ENABLE**  
+This variable specifies whether the specified service in SYSTEMD_SERVICE should start automatically or not. By default, the service is enabled to automatically start at boot time as follows: 
+```SYSTEMD_AUTO_ENABLE = "enable"```  
+>Note: you can disable this by setting the variable to ``` "disable"```
+4. **SYSTEMD_SERVICE**
+This variable specifies the systemd service name for a package.
+>Note:You can specify multiples services, each one separated by a space.
+```SYSTEMD_SERVICE:${PN} = "serviceName.service"```
+5. **SYSTEMD_PACKAGES**
+This variable locates the systemd unit files when they are not found in the main recipe's package.  
+By default, this variable is set such that the systemd unit files are assumed to reside in the recipes main package:
+```SYSTEMD_PACKAGES = "${PN}"```
+6. **FILES**  
+This variable provides a package name override that identifies the resulting package, then provide a space separated list of files or paths that identify the files you want included as part of the resulting package:
+```FILES:${PN} += "${systemd_unitdir}/system/serviceName.service"```
+## Example for systemd service recipe 
+
+```
```

Comment:
In general examples work better to explain things, but an example is not complete without an explanation